### PR TITLE
util(dm): update checking for fake rotate event

### LINKED ISSUE #11381: DM has wrong checking condition for fake roate event
### What did you do?

ref https://github.com/go-mysql-org/go-mysql/issues/893

### What did you expect to see?

_No response_

### What did you see instead?

_No response_

### Versions of the cluster

DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):
```console
(paste DM version here, and you must ensure versions of dmctl, DM-worker and DM-master are same)
```
Upstream MySQL/MariaDB server version:
```console
(paste upstream MySQL/MariaDB server version here)
```
Downstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):
```console
(paste TiDB cluster version here)
```
How did you deploy DM: tiup or manually?
```console
(leave TiUP or manually here)
```
Other interesting information (system version, hardware config, etc):
```console
>
>
```

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #11381

### What is changed and how it works?

make fake rotate event checking more strict

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by lyzx2001 — APPROVED** [review]

File: dm/relay/relay_writer_test.go [hunk 1]
```diff
@@ -18,23 +18,20 @@ import (
 	"os"
 	"path"
 	"path/filepath"
+	"testing"
 	"time"
 
 	gmysql "github.com/go-mysql-org/go-mysql/mysql"
 	"github.com/go-mysql-org/go-mysql/replication"
-	"github.com/pingcap/check"
```
Comment (lance6716) [inline dm/relay/relay_writer_test.go:25]: the file change is caused by switching testing framework

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**